### PR TITLE
add new partitioner

### PR DESCRIPTION
**Pull request by shlomihassan**
add new type of patittioner

### CODE REVIEW

File: core/src/main/java/io/confluent/connect/storage/util/DataUtils.java [hunk 1]
```diff
@@ -33,6 +33,13 @@ public static Object getField(Object structOrMap, String fieldName) {
       if (field == null) {
         throw new DataException(String.format("Unable to find nested field '%s'", fieldName));
       }
+      if (field == null) {
```
Comment (aviyehuda) [inline core/src/main/java/io/confluent/connect/storage/util/DataUtils.java]: It seems you accidentally copied this 3 lines

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        Object partitionKey = struct.get(fieldName);
+        Type type = valueSchema.field(fieldName).schema().type();
+        switch (type) {
+          case INT8:
+          case INT16:
+          case INT32:
+          case INT64:
+            Number record = (Number) partitionKey;
+            builder.append(fieldName + "=" + record.toString());
```
Comment (aviyehuda) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java]: the toString() here is not needed.  when you do "" + someObject ,  it automatically does "" + someObject.toString()  .

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+        Object partitionKey = struct.get(fieldName);
+        Type type = valueSchema.field(fieldName).schema().type();
+        switch (type) {
+          case INT8:
+          case INT16:
+          case INT32:
+          case INT64:
+            Number record = (Number) partitionKey;
```
Comment (aviyehuda) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java]: What is the reason you convert it to a number? if all you do is to do record.toString(), than there is no need to convert it. Unless you also want to validate its value.

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          Number record = (Number) fieldValue;
+          builder.append(record.toString());
+        } else if (fieldValue instanceof String) {
+          builder.append((String) fieldValue);
+        } else if (fieldValue instanceof Date) {
+          boolean booleanRecord = (boolean) fieldValue;
+          builder.append(Boolean.toString(booleanRecord));
+        } else {
+          log.error(
+                  "Unsupported type '{}' for user-defined timestamp field.",
```
Comment (jeremydonahue) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java]: > timestamp field
This should probably be _"field value"_.

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                  fieldValue.getClass()
+          );
+          throw new PartitionException(
+                  "Error extracting timestamp from record field: " + fieldName
```
Comment (jeremydonahue) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java]: > Error extracting timestamp from...
This should probably be "Error extracting _value_ from..."

File: core/src/main/java/io/confluent/connect/storage/util/DataUtils.java [hunk 6]
```diff
@@ -33,6 +33,7 @@ public static Object getField(Object structOrMap, String fieldName) {
       if (field == null) {
         throw new DataException(String.format("Unable to find nested field '%s'", fieldName));
       }
+      return field;
```
Comment (cricket007) [inline core/src/main/java/io/confluent/connect/storage/util/DataUtils.java]: This was fixed in #63

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+  }
+
+  @Override
+  public String encodePartition(SinkRecord sinkRecord) {
+    String timePartition = super.encodePartition(sinkRecord);
+
+    Object value = sinkRecord.value();
+    log.info(value.getClass().toString());
+
+    if (value instanceof Struct) {
```
Comment (atuljha-optimizely) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java]: Would `FieldPartitioner` be a better place for the field extraction logic, then we could simply return `fieldPartitioner.encodePartition(sinkRecord) + delim + super.encodePartition(sinkRecord)`

**Pull request closed without merging**